core/src/test/scala/kanaloa/reactive/dispatcher/IntegrationTests.scala

```diff
@@ -154,6 +170,55 @@ class PullingDispatcherIntegration extends IntegrationSpec {
 
 }
 
+class PullingDispatcherSanityCheckIntegration extends IntegrationSpec {
+
+  "can remain sane when all workers are failing" in new TestScope with Backends {
+    val backend = suicidal(1.milliseconds)
+    val iterator = Stream.continually("a").iterator
+
+    val pd = system.actorOf(PullingDispatcher.props(
+      "test-pulling",
+      iterator,
+      backend,
+      None,
+      ConfigFactory.parseString(
+        s"""
+          kanaloa.dispatchers.test-pulling {
+           updateInterval = 100ms
+            workerPool {
+              startingPoolSize = 30
+              minPoolSize = 30
+            }
+            $metricsConfig
+          }
+        """
+      )
+    )(resultChecker))
+
+    watch(pd)
+
+    val prob = TestProbe()
+    pd ! SubscribePerformanceMetrics(prob.ref)
```

Comment:
Not the biggest fan of exposing this via the `Dispatcher`, but i cannot think of a better way of doing it right now.